tests/signal_workflow.go

```diff
@@ -1563,21 +1563,30 @@ func (s *FunctionalSuite) TestSignalWithStartWorkflow() {
 	s.ProtoEqual(signalInput, signalEvent.GetWorkflowExecutionSignaledEventAttributes().Input)
 	s.Equal(identity, signalEvent.GetWorkflowExecutionSignaledEventAttributes().Identity)
 
-	// Assert visibility is correct
 	listOpenRequest := &workflowservice.ListOpenWorkflowExecutionsRequest{
 		Namespace:       s.namespace,
 		MaximumPageSize: 100,
 		StartTimeFilter: &filterpb.StartTimeFilter{
 			EarliestTime: nil,
 			LatestTime:   timestamppb.New(time.Now().UTC()),
 		},
-		Filters: &workflowservice.ListOpenWorkflowExecutionsRequest_ExecutionFilter{ExecutionFilter: &filterpb.WorkflowExecutionFilter{
-			WorkflowId: id,
-		}},
+		Filters: &workflowservice.ListOpenWorkflowExecutionsRequest_ExecutionFilter{
+			ExecutionFilter: &filterpb.WorkflowExecutionFilter{
+				WorkflowId: id,
+			},
+		},
 	}
-	listResp, err := s.engine.ListOpenWorkflowExecutions(NewContext(), listOpenRequest)
-	s.NoError(err)
-	s.Equal(1, len(listResp.Executions))
+
+	// Assert visibility is correct
+	s.Eventually(
+		func() bool {
+			listResp, err := s.engine.ListOpenWorkflowExecutions(NewContext(), listOpenRequest)
+			s.NoError(err)
+			return len(listResp.Executions) == 1
+		},
+		waitForESToSettle,
+		100*time.Millisecond,
+	)
```

Comment:
not sure what the philosophy is here about combining multiple changes into the same PR. Maybe this is fine because `s.Eventually()` is a clear improvement and doesnt change the idea of the code.

Reply:
I'll split the PR in two.

Reply:
https://github.com/temporalio/temporal/pull/5290